plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -2275,6 +2313,78 @@ public void finishStatisticsCollection(ConnectorSession session, ConnectorTableH
                 computedStatistics);
     }
 
+    private void generateMissingStatistics(ConnectorSession session, DeltaLakeTableHandle tableHandle, Collection<ComputedStatistics> computedStatistics)
+    {
+        Map<String, AddFileEntry> addFileEntriesWithNoStats = metastore.getValidDataFiles(tableHandle.getSchemaTableName(), session).stream()
+                .filter(addFileEntry -> addFileEntry.getStats().isEmpty()
+                        || addFileEntry.getStats().get().getMaxValues().isEmpty()
+                        || addFileEntry.getStats().get().getMinValues().isEmpty()
+                        || addFileEntry.getStats().get().getNullCount().isEmpty())
+                // Statistics returns whole path to file, so we need to create corresponding map key for AddFileEntry.
+                .map(addFileEntry -> Map.entry(appendPath(tableHandle.getLocation(), addFileEntry.getPath()), addFileEntry))
+                .collect(Collectors.toUnmodifiableMap(Entry::getKey, Entry::getValue));
+
+        if (addFileEntriesWithNoStats.isEmpty()) {
+            return;
+        }
+
+        Map<String, ComputedStatistics> groupedStatistics = Statistics.createComputedStatisticsToPartitionMap(computedStatistics, ImmutableList.of(PATH_COLUMN_NAME), ImmutableMap.of(PATH_COLUMN_NAME, VARCHAR))
```

Comment:
static import `createComputedStatisticsToPartitionMap`

Reply:
Usage of `createComputedStatisticsToPartitionMap` removed